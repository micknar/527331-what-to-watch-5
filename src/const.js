export const filmsCount = {
  SIMILAR: 4,
  PER_STEP: 8,
};

export const RatingMarkType = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const FilmPageNav = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  FILMS: `/films/`,
  FILMS_ID: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  REVIEW: `/films/:id/review`,
  ROOT: `/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id`,
};

export const APIRoute = {
  COMMENTS: `/comments/`,
  FILMS: `/films/`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
};
