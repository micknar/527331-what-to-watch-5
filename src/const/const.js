export const HOUR = 60;
export const WHITE_RGB_COLOR_CODE = 255;
export const CARD_HOVER_TIMEOUT = 1000;

export const RATING_MARKS = [1, 2, 3, 4, 5];

export const ratingMarkMap = {
  'bad': {
    text: `Bad`,
    value: 3,
  },
  'normal': {
    text: `Normal`,
    value: 5,
  },
  'good': {
    text: `Good`,
    value: 8,
  },
  'veryGood': {
    text: `Very good`,
    value: 10,
  },
  'awesome': {
    text: `Awesome`,
    value: 10,
  },
};

export const genreMap = {
  'comedy': {
    single: `Comedy`,
    plural: `Comedies`,
  },
  'drama': {
    single: `Drama`,
    plural: `Dramas`,
  },
  'thriller': {
    single: `Thriller`,
    plural: `Thrillers`,
  },
};

export const FilmsCount = {
  SIMILAR: 4,
  PER_STEP: 8,
};

export const FilmPageNav = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  FILMS: `/films/`,
  FILMS_ID: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  REVIEW: `/films/:id/review`,
  ROOT: `/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id`,
};

export const APIRoute = {
  COMMENTS: `/comments/`,
  FAVORITE_FILMS: `/favorite`,
  FILMS: `/films/`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
};

export const FavoriteStatusCode = {
  ADD: 1,
  REMOVE: 0,
};

export const NameSpace = {
  APP_STATE: `APP_STATE`,
  USER: `USER`,
};

export const Radix = {
  DECIMAL: 10,
  HEXADECIMAL: 16,
};

export const HttpCode = {
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  NOT_FOUND: 404,
};
