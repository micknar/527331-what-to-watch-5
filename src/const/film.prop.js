import PropTypes from "prop-types";

export default PropTypes.shape({
  id: PropTypes.number,
  name: PropTypes.string,
  posterImage: PropTypes.string,
  previewImage: PropTypes.string,
  backgroundImage: PropTypes.string,
  backgroundColor: PropTypes.string,
  videoLink: PropTypes.string,
  previewVideoLink: PropTypes.string,
  description: PropTypes.string,
  rating: PropTypes.number,
  scoresCount: PropTypes.number,
  ratingMark: PropTypes.string,
  director: PropTypes.string,
  starring: PropTypes.arrayOf(PropTypes.string),
  runTime: PropTypes.shape({
    hours: PropTypes.number,
    minutes: PropTypes.number,
  }),
  genre: PropTypes.string,
  released: PropTypes.number,
  isFavorite: PropTypes.bool,
});
