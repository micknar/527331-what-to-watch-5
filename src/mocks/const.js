export const FILMS_COUNT = 17;
export const FILMS_COUNT_PER_STEP = 8;
export const RATING_MIN = 0;
export const RATING_MAX = 10;
export const DIRECTORS = [`Anthony Mann`, `Martin Scorsese`, `Clint Eastwood`, `Ilya Naishuller`, `Guy Ritchie`];
export const ACTORS = [`Erich von Stroheim`, `Mary Beth Hughes`, `Dan Duryea`, `Robert Duvall`, `Liam Neeson`, `Russell Crowe`, `Woody Harrelson`, `Bill Murray`, `Edward Norton`, `Willem Dafoe`, `Tilda Swinton`];
export const USERS = [`John Doe`, `Tim Macoveev`, `Bill Murray`, `Jude Law`, `Tony Revoloru`, `Owen Wilkinson`, `Jeff Goldblum`];

export const GENRES = [
  `Comedy`,
  `Crime`,
  `Documentary`,
  `Drama`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thriller`,
];

export const TITLES = [
  `Aviator`,
  `Bohemian Rhapsody`,
  `Dardjeeling Limited`,
  `Fantastic Beasts The Crimes Of Grindelwald`,
  `Johnny English`,
  `Macbeth`,
  `Midnight Special`,
  `Mindhunter`,
];

export const POSTERS = [
  `/img/aviator.jpg`,
  `/img/bohemian-rhapsody.jpg`,
  `/img/dardjeeling-limited.jpg`,
  `/img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `/img/johnny-english.jpg`,
  `/img/macbeth.jpg`,
  `/img/midnight-special.jpg`,
  `/img/mindhunter.jpg`,
];

export const TEXTS = [
  `Lorem ipsum dolor sit amet, consectetur adipiscing elit.`,
  `Cras aliquet varius magna, non porta ligula feugiat eget.`,
  `Fusce tristique felis at fermentum pharetra.`,
  `Aliquam id orci ut lectus varius viverra.`,
  `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
  `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
  `Sed sed nisi sed augue convallis suscipit in sed felis.`,
  `Aliquam erat volutpat.`,
  `Nunc fermentum tortor ac porta dapibus.`,
  `In rutrum ac purus sit amet tempus.`,
];
