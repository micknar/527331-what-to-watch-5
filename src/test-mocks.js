import {getDate} from "./utils";

export const noop = () => { };

export const films = [
  {
    id: 1,
    name: `Bronson`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `/img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 8.9,
    ratingMark: `Booring`,
    scoresCount: 240,
    director: `Nicolas Winding Refn`,
    starring: [`Bill Murray`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 40,
    },
    genre: `Crime`,
    released: 1996,
    isFavorite: false,
  },
  {
    id: 2,
    name: `Midnight Special`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Midnight_Special.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 10,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 42,
    },
    genre: `Thriller`,
    released: 1989,
    isFavorite: true,
  },
  {
    id: 3,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Pulp_Fiction.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amo eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 2.2,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 22,
    },
    genre: `Action`,
    released: 2007,
    isFavorite: false,
  },
  {
    id: 4,
    name: `Matrix`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Matrix.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ext enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.5,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 24,
    },
    genre: `Adventure`,
    released: 2011,
    isFavorite: true,
  },
  {
    id: 5,
    name: `Legend`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Legend.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem i ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.1,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 57,
    },
    genre: `Comedy`,
    released: 2015,
    isFavorite: false,
  },
  {
    id: 6,
    name: `angs of new york`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/gangs_of_new_york.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sedrem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.uip ex ea commodo consequat.`,
    rating: 7.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 35,
    },
    genre: `Fantasy`,
    released: 1980,
    isFavorite: false,
  },
  {
    id: 7,
    name: `A Star Is Born`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_is_Born.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_is_Born.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 6.4,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Guy Ritchie`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 25,
    },
    genre: `Drama`,
    released: 1989,
    isFavorite: false,
  },
  {
    id: 8,
    name: `Snatch`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Snatch.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliqliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 9.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 20,
    },
    genre: `Fantasy`,
    released: 2004,
    isFavorite: false,
  },
  {
    id: 9,
    name: `Bohemian Rhapsody`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Bohemian_Rhapsody.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Bohemian_Rhapsody.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur.`,
    rating: 3,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: {
      hours: 1,
      minutes: 20,
    },
    genre: `Adventure`,
    released: 1998,
    isFavorite: false,
  },
];

export const comments = [
  {
    user: {
      id: 19,
      name: `Christina`,
    },
    rating: 2.4,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 3.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-27T13:38:44.769Z`),
  },
  {
    user: {
      id: 2,
      name: `Mollie`,
    },
    rating: 1.7,
    comment: `Lorem ipsustrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 4.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ilaboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-11T13:38:44.769Z`),
  },
  {
    user: {
      id: 11,
      name: `Jack`,
    },
    rating: 4.5,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitatboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-19T13:38:44.769Z`),
  },
];
