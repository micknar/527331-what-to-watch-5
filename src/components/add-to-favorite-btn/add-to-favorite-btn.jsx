import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {AppRoute, AuthorizationStatus} from "../../const/const";
import {updateFavoriteStatus} from "../../store/api-actions";
import {setIsFilmLoading} from "../../store/action";

const AddToFavoriteBtn = (props) => {
  const {id, isFavorite, authorizationStatus, onAddToFavoriteBtnClick} = props;

  return (
    <Link
      to={authorizationStatus !== AuthorizationStatus.AUTH ? AppRoute.LOGIN : ``}
      className="btn btn--list movie-card__button"
      onClick={(evt) => {
        if (authorizationStatus === AuthorizationStatus.AUTH) {
          evt.preventDefault();
          onAddToFavoriteBtnClick(id, isFavorite);
        }
      }}
    >
      {
        isFavorite
          ?
          <svg viewBox="0 0 18 14" width="18" height="14">
            <use xlinkHref="#in-list" />
          </svg>

          :
          <svg viewBox="0 0 19 20" width="19" height="20">
            <use xlinkHref="#add" />
          </svg>
      }

      <span>My list</span>
    </Link>
  );
};

AddToFavoriteBtn.propTypes = {
  id: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  onAddToFavoriteBtnClick: PropTypes.func.isRequired,
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
});

const mapDispatchToProps = (dispatch) => ({
  onAddToFavoriteBtnClick(id, isFavorite) {
    dispatch(setIsFilmLoading(true));
    dispatch(updateFavoriteStatus(id, isFavorite));
  }
});

export {AddToFavoriteBtn};
export default connect(mapStateToProps, mapDispatchToProps)(AddToFavoriteBtn);
