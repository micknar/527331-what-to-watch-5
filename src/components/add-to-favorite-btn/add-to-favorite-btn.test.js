import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {AddToFavoriteBtn} from "./add-to-favorite-btn";
import {noop} from "../../test-mocks";
import {AuthorizationStatus} from "../../const/const";

describe(`Should AddToFavoriteBtn render correctly`, () => {
  it(`When authorized user adds to favorites`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <AddToFavoriteBtn
              id={1}
              isFavorite={false}
              authorizationStatus={AuthorizationStatus.AUTH}
              onAddToFavoriteBtnClick={noop}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When authorized user remove from favorites`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <AddToFavoriteBtn
              id={1}
              isFavorite={true}
              authorizationStatus={AuthorizationStatus.AUTH}
              onAddToFavoriteBtnClick={noop}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When guest clicks on 'MyList' button`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <AddToFavoriteBtn
              id={1}
              isFavorite={false}
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              onAddToFavoriteBtnClick={noop}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
