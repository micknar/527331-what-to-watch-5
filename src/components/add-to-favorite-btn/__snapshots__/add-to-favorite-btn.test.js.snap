// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should AddToFavoriteBtn render correctly When authorized user adds to favorites 1`] = `
<a
  className="btn btn--list movie-card__button"
  href="/"
  onClick={[Function]}
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</a>
`;

exports[`Should AddToFavoriteBtn render correctly When authorized user remove from favorites 1`] = `
<a
  className="btn btn--list movie-card__button"
  href="/"
  onClick={[Function]}
>
  <svg
    height="14"
    viewBox="0 0 18 14"
    width="18"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</a>
`;

exports[`Should AddToFavoriteBtn render correctly When guest clicks on 'MyList' button 1`] = `
<a
  className="btn btn--list movie-card__button"
  href="/login"
  onClick={[Function]}
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</a>
`;
